__tests__/azure.azHttpClient.test.ts

```diff
@@ -0,0 +1,275 @@
+        jest.spyOn(core, 'debug').mockImplementation();
+        const policyIds = [{
+            policy: 'providers/Microsoft.Management/managementgroups/abcdef/providers/Microsoft.Authorization/policySetDefinitions/12',
+
+        }, {
+            policy: 'providers/Microsoft.Management/managementgroups/abcdef/providers/Microsoft.Authorization/policySetDefinitions/78',
+
+        }] as PolicyRequest[];
+        
+        expect(await testClient.upsertPolicyDefinitions(policyIds)).toMatchObject([{}, {}]);
+    });
+
+    test('upsertPolicyDefinitions() - throw if response doesn\'t have same length', async () => {
+        jest.spyOn(httpClient, 'sendRequest').mockResolvedValue({
+            statusCode: 200,
+            statusMessage: 'success',
+            headers: {},
+            body: {
+                responses: [{}, {}]
+            }
+        });
+        jest.spyOn(console, 'log').mockImplementation();
+        jest.spyOn(core, 'debug').mockImplementation();
+        const policyIds = [{
+            policy: 'providers/Microsoft.Management/managementgroups/abcdef/providers/Microsoft.Authorization/policySetDefinitions/12',
+        }] as PolicyRequest[];
+        
+        await expect(testClient.upsertPolicyDefinitions(policyIds)).rejects.toThrow('Azure batch response count does not match batch request count');
+    });
+
+    test('upsertPolicyInitiatives() - insert or update policy definition', async () => {
+        jest.spyOn(httpClient, 'sendRequest').mockResolvedValue({
+            statusCode: 200,
+            statusMessage: 'success',
+            headers: {},
+            body: {
+                responses: [{}, {}]
+            }
+        });
+        const policyIds = [{
+            policy: 'providers/Microsoft.Management/managementgroups/abcdef/providers/Microsoft.Authorization/policySetDefinitions/12',
+
+        }, {
+            policy: 'providers/Microsoft.Management/managementgroups/abcdef/providers/Microsoft.Authorization/policySetDefinitions/78',
+
+        }] as PolicyRequest[];
+        
+        expect(await testClient.upsertPolicyDefinitions(policyIds)).toMatchObject([{}, {}]);
+    });
+
+    test('deletePolicies() - delete a policy using id', async () => {
+        jest.spyOn(httpClient, 'sendRequest').mockResolvedValue({
+            statusCode: 200,
+            statusMessage: 'success',
+            headers: {},
+            body: {
+                responses: [{}, {}]
+            }
+        });
+        const policyIds = [
+            'providers/Microsoft.Management/managementgroups/abcdef/providers/Microsoft.Authorization/policySetDefinitions/12',
+            'providers/Microsoft.Management/managementgroups/abcdef/providers/Microsoft.Authorization/policySetDefinitions/78',
+        ];
+        
+        expect(await testClient.deletePolicies(policyIds)).toMatchObject([{}, {}]);
+    });
+   
+    test('deletePolicies() - throw if response doesn\'t have same length', async () => {
```

Comment:
Something clearer such as - 'throw if number of response objects is not equal to number of request objects' ?